packages/emotion/src/useStyle.ts

```diff
@@ -98,18 +102,21 @@ const useStyle = <
     typeof generateComponentTheme === 'function'
       ? generateComponentTheme(theme as Theme)
       : {}
+  const componentWithTokensId = useTokensFrom ?? componentId
```

Comment:
Add it here as a cast, e.g. `(useStyleParams as UseStyleParamsNew<P>).useTokensFrom`